subsys/net/lib/lwm2m/lwm2m_shell.c

```diff
@@ -176,7 +177,17 @@ static int cmd_read(const struct shell *sh, size_t argc, char **argv)
 	if (argc > 2) { /* read + path + options(data type) */
 		dtype = argv[2];
 	}
-	if (strcmp(dtype, "-s") == 0) {
+	if (strcmp(dtype, "-x") == 0) {
+		const char *buff;
+		uint16_t buff_len = 0;
+
+		ret = lwm2m_get_res_buf(&path, (void **)&buff,
+					&buff_len, NULL, NULL);
```

Comment:
```suggestion
		ret = lwm2m_get_res_buf(&path, (void **)&buff,
					NULL, &buff_len, NULL);
```

Please use the data length, instead of buffer length.
There is no point of printing the full buffer, when there is less data stored.

Reply:
Good catch!
What about the string formatter which does the same? Looks rather unsafe either. IMHO that should be changed also.

```
ret = lwm2m_engine_get_res_buf(pathstr, (void **)&buff, NULL, &buff_len, NULL);
if (ret != 0) {
	goto out;
}
shell_print(sh, "%.*s", buff_len, buff);
```

Reply:
Yes, please change that as well.
Strings are zero-terminated so it should not print completely bogus, but still it should be pedantic.